crates/ethrpc/src/block_stream/mod.rs

```diff
@@ -252,116 +247,104 @@ pub trait BlockRetrieving: Debug + Send + Sync + 'static {
     async fn current_block(&self) -> Result<BlockInfo> {
-        get_block_at_id(self, BlockNumber::Latest.into())
-            .await?
-            .try_into()
+        get_block_at_id(self, BlockId::latest()).await?.try_into()
     }
 
     async fn block(&self, number: u64) -> Result<BlockNumberHash> {
-        let block = get_block_at_id(self, U64::from(number).into()).await?;
-        Ok((
-            block.number.context("missing block_number")?.as_u64(),
-            block.hash.context("missing block_hash")?,
-        ))
+        let block = get_block_at_id(self, BlockId::number(number)).await?;
+        Ok((block.header.number, block.header.hash.into_legacy()))
     }
 
     /// Gets all blocks requested in the range. For successful results it's
     /// enforced that all the blocks are present, in the correct order and that
     /// there are not reorgs in the block range.
     async fn blocks(&self, range: RangeInclusive<u64>) -> Result<Vec<BlockNumberHash>> {
-        let include_txs = helpers::serialize(&false);
         let (start, end) = range.clone().into_inner();
-        let mut batch_request = Vec::with_capacity((end - start + 1) as usize);
-        for i in start..=end {
-            let num = helpers::serialize(&BlockNumber::Number(i.into()));
-            let request = self
-                .transport()
-                .prepare("eth_getBlockByNumber", vec![num, include_txs.clone()]);
-            batch_request.push(request);
-        }
-
-        let mut prev_hash = None;
 
-        // send_batch guarantees the size and order of the responses to match the
-        // requests
-        self.transport()
-            .send_batch(batch_request.iter().cloned())
-            .await?
-            .into_iter()
-            .map(|response| match response {
-                Ok(response) => {
-                    serde_json::from_value::<web3::types::Block<H256>>(response.clone())
-                        .with_context(|| format!("unexpected response format: {response:?}"))
-                        .and_then(|response| {
-                            let current_hash = response.hash.context("missing hash")?;
-                            let current_block = response.number.context("missing number")?.as_u64();
-                            if prev_hash.is_some_and(|prev| prev != response.parent_hash) {
-                                tracing::debug!(
-                                    ?range,
-                                    ?prev_hash,
-                                    parent_hash = ?response.parent_hash,
-                                    "inconsistent parent in block range"
-                                );
-                                return Err(anyhow!("inconsistent block range"));
-                            }
-                            prev_hash = Some(current_hash);
-
-                            Ok((current_block, current_hash))
-                        })
+        let block_futures: Vec<_> = (start..=end)
```

Comment:
this feels like a code smell, cloning the range to use it for debug and then reconstructing it manually for actual "range-y" things. it's a good way of someone shooting themselves on their foot by removing an `=`